include/nbl/asset/ICPUMeshBuffer.h

```diff
@@ -57,580 +58,748 @@ namespace impl
     }
 }
 
-class ICPUMeshBuffer : public IMeshBuffer<ICPUBuffer, ICPUDescriptorSet, ICPURenderpassIndependentPipeline>, public BlobSerializable, public IAsset
+class ICPUMeshBuffer final : public IMeshBuffer<ICPUBuffer,ICPUDescriptorSet,ICPURenderpassIndependentPipeline,ICPUSkeleton>, public BlobSerializable, public IAsset
 {
-    using base_t = IMeshBuffer<ICPUBuffer, ICPUDescriptorSet, ICPURenderpassIndependentPipeline>;
-    //vertices
-    uint32_t posAttrId = 0u;
-    uint32_t normalAttrId = MAX_VERTEX_ATTRIB_COUNT;
+        using base_t = IMeshBuffer<ICPUBuffer,ICPUDescriptorSet,ICPURenderpassIndependentPipeline,ICPUSkeleton>;
+        // knowing the position attribute ID is important for AABB computations etc.
+        uint32_t posAttrId : 5;
+        uint32_t normalAttrId : 5;
+        // by having one attribute only, we limit the number of bones per vertex to 4
+        uint32_t jointIDAttrId : 5;
+        uint32_t jointWeightAttrId : 5;
```

Comment:
remember to include those new members in `duplicateMeshbuffer()` (or something like this) in mesh manipulator

Reply:
its gone, it pissed me off